frontend/wallet/src/render/views/footer/SendButton.re

```diff
@@ -1,217 +1,54 @@
-  let (sendState, setModalState) = React.useState(_ => None);
-  let activeWallet = Hooks.useActiveWallet();
-  let spacer = <Spacer height=0.5 />;
-  ModalState.Unvalidated.(
-    <>
-      <Button
-        label="Send"
-        onClick={_ => setModalState(_ => Some(emptyModal(activeWallet)))}
-      />
-      {switch (sendState) {
-       | None => React.null
-       | Some(
-           {fromStr, toStr, amountStr, feeStr, memoOpt, errorOpt} as fullState,
-         ) =>
-         <Modal
-           title="Send Coda" onRequestClose={() => setModalState(_ => None)}>
-           <div className=Styles.contentContainer>
-             {switch (errorOpt) {
-              | None => React.null
-              | Some(err) => <Alert kind=`Danger message=err />
-              }}
-             spacer
-             <Dropdown
-               label="From"
-               value=fromStr
-               onChange={value =>
-                 setModalState(
-                   Option.map(~f=s => {...s, fromStr: Some(value)}),
-                 )
-               }
-               options={
-                 wallets
-                 |> Array.map(~f=wallet =>
-                      (
-                        PublicKey.toString(wallet.Wallet.publicKey),
-                        <WalletDropdownItem wallet />,
-                      )
-                    )
-                 |> Array.toList
-               }
-             />
-             spacer
-             <TextField
-               label="To"
-               mono=true
-               onChange={value =>
-                 setModalState(Option.map(~f=s => {...s, toStr: value}))
-               }
-               value=toStr
-               placeholder="Recipient Public Key"
-             />
-             spacer
-             <TextField.Currency
-               label="QTY"
-               onChange={value =>
-                 setModalState(Option.map(~f=s => {...s, amountStr: value}))
-               }
-               value=amountStr
-               placeholder="0"
-             />
-             spacer
-             <TextField.Currency
-               label="Fee"
-               onChange={value =>
-                 setModalState(Option.map(~f=s => {...s, feeStr: value}))
+    {switch (activeWallet) {
+     | None => React.null
+     | Some(pubkey) =>
+       let walletQuery =
+         WalletLocked.make(~publicKey=Apollo.Encoders.publicKey(pubkey), ());
+       <WalletQuery variables=walletQuery##variables>
+         (
+           response =>
+             switch (response.result) {
+             | Loading
+             | Error(_) => React.null
+             | Data(data) =>
+               switch (modalOpen, data##wallet) {
+               | (false, _)
+               | (true, None) => React.null
+               | (true, Some(wallet)) =>
+                 wallet##locked
+                   ? <UnlockModal
+                       wallet=pubkey
+                       onClose={() => setModalOpen(_ => false)}
```

Comment:
Does onClose get triggered when unlock is successful or does it transition to the send flow from there?

Reply:
Maybe we need to add a "closeOnUnlock" prop to the UnlockModal to support this?